models/remoteworkitem.go

```diff
@@ -0,0 +1,21 @@
+package models
+
+// RemoteWorkItem represents a work item as it is stored in the database
+type RemoteWorkItem struct {
+	Fields map[string]interface{}	
+}
+
+// The intregration code for a specific remote provider must populate this.
+type WorkItemMap map[string]interface{}
+
+// Map maps RemoteWorkItem to WorkItem
+func (rwi RemoteWorkItem) MapRemote (wiMap WorkItemMap, wiType WorkItemType) WorkItem {
+	workItem := WorkItem{
+		Type: wiType.ID
+	}	
+	
+	for from_key, to_key := range wiMap {
+		workItem.Fields[to_key] = remoteWorkItem.Fields[from_key]
```

Comment:
Thanks @aslakknutsen .
This is how I plan to implement the multi-level key access:

The code which pulls the data from the remote provider would take care of flattening everything to level 1 ( I can't seem to think of a use case where this is not possible ).  

```
issue: {
  meta: {
    title: ""
  }
}
```  

should be sanitized/converted to a dict of :
```
{
   "title" : "some title",
  "description" : "some desc"
}
```
The flattened dict would be assigned to type `RemoteWorkItem`.

By doing the above, we can ensure that the mapping code doesn't change, and also that every provider will have it's own format so it should be best left to the integration code to standardize the 
structure into a Flat 1 evel.